lib/std/Build/Step/Compile.zig

```diff
@@ -817,17 +824,20 @@ pub fn linkFrameworkWeak(c: *Compile, name: []const u8) void {
 
 /// Handy when you have many C/C++ source files and want them all to have the same flags.
 pub fn addCSourceFiles(compile: *Compile, options: Module.AddCSourceFilesOptions) void {
+    assert(compile.kind != .pch); // pch can only be generated from a single C header file
```

Comment:
I feel it's a bit odd for `compile.addCSourceFiles` (and the similar functions below) to have different behavior than `compile.root_module.addCSourceFiles`, especially considering the latter is invalid as well and would probably crash on the assert in finalize. Additionally, it seems most of the functions in `Step.Compile` don't make much sense for a precompiled header. It may be worth just making a seperate `Step.Compile.Pch` or `Step.PrecompileHeader` similar to what `Step.TranslateC` does.

Reply:
> Additionally, it seems most of the functions in Step.Compile don't make much sense for a precompiled header.

mmm I don't  _most_ .  The precompiled header are sensitive to most compiler options, so for instance even "linkXXX" functions may change compile flags and defines that impact the header compilation even though the actual linking doesn't take place.

Reply:
I wouldn't expect any linker options to affect a precompiled header since linking would be done by the executable/library Compile step, although you might know better than I. Also, I can't think of much use for the functions below:
```
installHeader*
addObjCopy
checkObject
setLinkerScript, setVersionScript // only for the linker
forceUndefinedSymbol // for the linker, doesn't really make sense for a pch
isDynamicLibrary/isStaticLibrary/isDll/producesPdbFile/producesImplib // always false
link* // shouldn't change the output, unless pch's fail to compile when the symbols aren't resolved.
addCSourceFile* // can't compile anything more than the pch
addWin32ResourceFile
setVerboseLink // no linking takes place, therefore no output from the linker
getEmitted* //  most of these would fail assertions on a pch
addObjectFile/addObject
```
That's a large amount of the user-facing API of Step.Compile which wouldn't make sense to use in this context, I think it's worth having a separate step so that the API makes it more clear on what affects the compilation of the PCH, what is impossible/doesn't make sense, and what doesn't affect compilation.